lib/modules/dosomething/dosomething_signup/includes/SignupsController.php

```diff
@@ -2,6 +2,8 @@
 
 class SignupsController extends EntityAPIController {
 
+  protected $ignore_global_user = FALSE;
```

Comment:
Yeah, I think it'd make sense to add a new parameter to the `save()` method for that rather than relying on `global $user` in there. Having to switch modes like this feels like it'd be easy to make a mistake with.